ehr/src/org/labkey/ehr/utils/TriggerScriptHelper.java

```diff
@@ -1458,14 +1459,61 @@ public void updateParentRequestQcState(String requestId, Integer qcStateId)
         }
     }
 
+    // Get all datasets that have a requestid column
+    private List<TableInfo> getRequestDatasets(Container c, User u)
+    {
+        Study study = StudyService.get().getStudy(c);
+        if (null == study)
+        {
+            throw new IllegalStateException("Study not found for container: " + c.getName());
+        }
+
+        List<? extends Dataset> datasets = study.getDatasets();
+        List<TableInfo> requestDatasets = new ArrayList<>();
+        for (Dataset ds : datasets)
+        {
+            TableInfo ti = StorageProvisioner.createTableInfo(ds.getDomain());  // This is faster than ds.getTableInfo(u)
+            if (ti.getColumn("requestid") != null)
+            {
```

Comment:
Even faster to avoid the `TableInfo` when we don't need it.

```suggestion
            if (ds.getDomain().getPropertyByName("requestid") != null)
            {
                TableInfo ti = StorageProvisioner.createTableInfo(ds.getDomain());  // This is faster than ds.getTableInfo(u)
```

Reply:
Added